### ISSUE TITLE
Edit item doesn't work in custom listview layout using listViewHelper

### ISSUE THREAD

**Issue by bjarnef** [umbraco/Umbraco-CMS#10016]
It seems current implementation of `listViewHelper.editItem(item);` doesn't work correct after upgrade to Umbraco v8.12.1
I seems it was affected by the changes in this PR: https://github.com/umbraco/Umbraco-CMS/pull/9026

In this case I upgraded a site from v8.8.4 to v.8.12.1 the "go to" link doesn't work in my custom listview layout. The "go to" links does work in the default list and grid layouts though.

```
$scope.gotoItem = function (item) {
    listViewHelper.editItem(item);
};
```

It is throwing this console error:
![image](https://user-images.githubusercontent.com/2919859/111779211-23bdd180-88b6-11eb-8231-c03aa2464bd1.png)

Add a custom listview layout, e.g "grid-employees" and place these files in a folder in `App_Plugins`: /App_Plugins/ListviewLayouts/grid-employees/grid.html

**Controller**
```
(function () {
    "use strict";

    function ListViewGridLayoutController($scope, $routeParams, mediaHelper, mediaResource, listViewHelper) {

        var vm = this;

        vm.nodeId = $scope.contentId;

        $scope.toggleItem = function (item) {
            if (item.selected) {
                listViewHelper.deselectItem(item, $scope.selection);
            } else {
                listViewHelper.selectItem(item, $scope.selection);
            }
        };

        $scope.gotoItem = function (item) {
            listViewHelper.editItem(item);
        };

        $scope.items = $scope.items.map(function (obj) {

            if (obj.contentTypeAlias === "employee") {
                var imageProp = _.find(obj.properties, function (p) { return (p.alias === "image"); });
                if (imageProp) {
                    obj.thumbnail = "https://www.gravatar.com/avatar/?d=mm&s=100&f=y";

                    var mediaId = imageProp.value;
                    if (mediaId) {
                        mediaResource.getById(mediaId).then(function (media) {
                            obj.thumbnail = mediaHelper.resolveFile(media) + "?width=100&height=100&mode=crop&animationprocessmode=first";
                        });
                    }
                }
            }

            return obj;
        });
    }

    angular.module("umbraco").controller("ListView.GridMembersLayoutController", ListViewGridLayoutController);
})();
```

**View**
```
<div ng-controller="ListView.GridMembersLayoutController as vm">
    
    <div class="umb-content-grid employees-grid">     
        <div ng-repeat="item in items" 
             class="umb-content-grid__item employees-grid__item umb-outline umb-outline--surrounding"
             ng-class="{' -selected': item.selected }"
             ng-click="toggleItem(item)"
             role="button"
             tabindex="0">
            <div class="employee-details">
                <div class="image-wrapper">
                    <img ng-if="item.thumbnail" src="" ng-src="{{item.thumbnail}}" alt="" draggable="false" />
                    <umb-icon icon="{{item.icon}}"></umb-icon>
                </div>
                <a href="" ng-click="gotoItem(item)" class="employee-name">{{item.name}}</a>
                <small class="muted">{{item.email}}</small>
            </div>
        </div>
    </div>
    
    <umb-empty-state
        ng-if="!items && options.filter.length > 0"
        position="center">
        <localize key="general_searchNoResult"></localize>
    </umb-empty-state>

</div>
```

Furthermore configurate the listview:

![image](https://user-images.githubusercontent.com/2919859/111780640-f2de9c00-88b7-11eb-9da8-13caa8ba2320.png)

In the case the document type is "Employee" (alias "employee").

The "Image" property using "Media Picker" datatype, "Phone" is a regular Textstring datatype and "Email" is using  "Email Address" datatype.

---
_This item has been added to our backlog [AB#11003](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/11003)_

**Comment by nul800sebastiaan** [comment]
Thanks @bjarnef - we'll have to have a look.

**Comment by bjarnef** [comment]
I think the issue is here where it expect `scope` to be passed in, which wasn't an requirement previous :)
https://github.com/umbraco/Umbraco-CMS/pull/9026/files#diff-b25bc893abf8a667b5dfb384d7a8c103f74780a8ce1c2e8c8c33c89247cd7d24R581

**Comment by bjarnef** [comment]
PR submitted: https://github.com/umbraco/Umbraco-CMS/pull/10018 🎉

**Comment by nathanwoulfe** [comment]
This is fixed in #10018
